src/cli/args.rs

```diff
@@ -44,6 +44,10 @@ pub struct CliArgs {
     /// Ouch and claps subcommands
     #[command(subcommand)]
     pub cmd: Subcommand,
+
+    /// Compression raw level as each algo has
+    #[arg(short = 'r', long, global = true, default_value_t = u32::MAX)]
+    pub raw_level: u32,
```

Comment:
We should use `Option` because `u32::MAX` doesn't mean anything, and we can use things like `map_or_else(Default::default, gzp::par::compress::Compression::new)` and avoid the repetitive `if` statements

`u16` so `as i32` can't panic, the compression levels should be 0-10 anyway

I'm also not sure about the name `--raw-level`, maybe just `--level` or `--compression-level`?

```suggestion
    pub level: Option<u16>,
```

Reply:
as to the clippy error, you can add an `allow`

Reply:
also this is specific to compression and shouldn't be a global flag

Reply:
Actually, there can be negative compression levels, don't ask me why xD.

https://en.wikipedia.org/wiki/Zstd#:~:text=It%20is%20tunable%20with%20compression,%2C%20but%20best%20compression%20ratio.

Reply:
so i16 would probably make more sense then